src/gluonts/ev/metrics.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass
+from functools import partial
+from typing import Optional
+
+import numpy as np
+
+from gluonts.exceptions import GluonTSUserError
+from .api import (
+    DerivedMetricEvaluator,
+    MetricEvaluator,
+    StandardMetricEvaluator,
+)
+from .aggregations import Mean, Sum
+from .stats import (
+    absolute_error,
+    absolute_label,
+    absolute_percentage_error,
+    absolute_scaled_error,
+    coverage,
+    quantile_loss,
+    scaled_interval_score,
+    squared_error,
+    symmetric_absolute_percentage_error,
+)
+
+
+def assert_axis_is_None_or_one(axis: Optional[int], metric_name: str) -> None:
+    if axis not in [None, 1]:
+        raise GluonTSUserError(
+            f"'{metric_name}' requires 'axis' to be None or 1 (not {axis})"
+        )
+
+
+@dataclass
+class AbsoluteLabelMean:
```

Comment:
Everything that has no parameters should be a function.

Reply:
So, something like this?

```
def absolute_label_mean():
    def get_absolute_mean_evaluator(axis: Optional[int] = None) -> MetricEvaluator:
        return StandardMetricEvaluator(
            map=absolute_label,
            aggregate=Mean(axis=axis),
        )

    return get_absolute_mean_evaluator
```

This would lead to inconsistency, some metrics being functions and others being classes. Not sure if if that's worth it...

Reply:
No, just

```py

def absolute_label_mean(axis: Optional[int] = None) -> MetricEvaluator:
    return StandardMetricEvaluator(
        map=absolute_label,
        aggregate=Mean(axis=axis),
    )
```

Reply:
> some metrics being functions and others being classes

What's the difference?

Reply:
The capitalization. 😄 But more importantly, I think it looks looks a bit strange to create objects for some metrics and use functions for others. For example:

```
metrics_per_entry = [MeanSquaredError(), QuantileLoss(), absolute_label_mean]
multi_metric = MultiMetricEvaluator()
multi_metric.add_metrics(metrics_per_entry, axis=1)
```

Reply:
The difference is that some have parameters while others don't.

Reply:
So, using it like above would be okay with you?

I found one other issue: If we make `absolute_label_mean` a function, than it doesn't satisfy the Metric Protocol anymore:
```
@runtime_checkable
class Metric(Protocol):
    def __call__(self, axis: Optional[int] = None) -> "MetricEvaluator":
        raise NotImplementedError
```

Is there something we can do here so that the Protocol satisfies both?

Another thought: We could also let all metrics be functions. It's a bit weird having them as dataclasses when their only purpose is to be called and return a `MetricEvaluator` - even if they take parameters. For metric that **don't** have any parameters, we'd end up with something like [above](https://github.com/awslabs/gluonts/pull/2267#discussion_r1001712848) - this also looks a bit strange but this way of writing things would be consistent for all metrics.

So, I lean more toward consistency here than differentiating between metrics with and without parameters. Still, I'm fine with changing it as you suggest.

Reply:
According to https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols it should work for both functions and classes.

I don't see how your approach would lead to more consistency. Is "but they are all classes" the consistent behaviour we are looking for?

We have defined a metric as something you can call with an `axis` parameter, which then returns an evaluator. It doesn't matter what that thing is, as long as it implements that interface, and functions do.

Reply:
Alright, you convinced me. 💯